### PR TITLE
feat: add snapshot info card

### PR DESCRIPTION
**Pull request by relaxgo**
#2721

### CODE REVIEW

File: packages/maskbook/src/plugins/Snapshot/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export function createETHAddressUrl(address: string): string {
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Snapshot/helpers.ts:1]: Move this helper function to `packages/maskbook/src/web3/pipes.ts`.

File: packages/maskbook/src/plugins/Snapshot/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    return `https://etherscan.io/address/${address}`
+}
+
+export function createIPFSUrl(ipfs: string): string {
+    return `https://ipfs.fleek.co/ipfs/${ipfs}`
+}
+
+export function createSnapshotUrl(snapshot: string): string {
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Snapshot/helpers.ts:9]: Move this helper function to `packages/maskbook/src/web3/pipes.ts`.

File: packages/maskbook/src/plugins/Snapshot/UI/SnapshotCard.tsx [hunk 3]
```diff
@@ -4,17 +4,31 @@ import { useI18N } from '../../../utils/i18n-next-ui'
 const useStyles = makeStyles((theme) => {
     return createStyles({
         root: {
+            width: '80%',
+            minHeight: 120,
+            padding: 0,
+            margin: '16px auto',
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Snapshot/UI/SnapshotCard.tsx:10]:
```suggestion
            margin: `${theme.spacing(2)} auto`,
```

File: packages/maskbook/src/plugins/Snapshot/UI/InformationCard.tsx [hunk 4]
```diff
@@ -1,20 +1,106 @@
...
+                                rel="noopener"
+                                href={createETHAddressUrl(strategy.params.address)}>
+                                <TokenIcon address={strategy.params.address} />
+                            </Link>
+                        ))}
+                    </Box>
+                </InfoField>
+                <InfoField title={t('plugin_snapshot_info_author')}>
+                    <Link className={classes.link} target="_blank" rel="noopener" href={createETHAddressUrl(author)}>
+                        {author.slice(0, 6)}...{author.slice(-4)}
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Snapshot/UI/InformationCard.tsx:82]: Use `formatEthereumAddress` instead.

File: packages/maskbook/src/plugins/Snapshot/UI/InformationCard.tsx [hunk 5]
```diff
@@ -1,20 +1,106 @@
...
+        },
+    })
 })
 
+function InfoField(props: InfoFieldProps) {
+    const classes = useStyles()
+
+    return (
+        <Box className={classes.field}>
+            <Box>{props.title}</Box>
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Snapshot/UI/InformationCard.tsx:43]: Plain text should be wrapped by `<Typography />`.

File: packages/maskbook/src/plugins/Snapshot/UI/SnapshotCard.tsx [hunk 6]
```diff
@@ -4,24 +4,25 @@ import { useI18N } from '../../../utils/i18n-next-ui'
 const useStyles = makeStyles((theme) => {
     return createStyles({
         root: {
-            margin: '16px auto',
-            border: `solid 1px ${theme.palette.divider}`,
+            minHeight: 120,
             padding: 0,
-            minHeight: 320,
+            border: `solid 1px ${theme.palette.divider}`,
+            margin: 'theme.spacing(2) auto',
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Snapshot/UI/SnapshotCard.tsx:10]: You need string template here.

**Review by guanbinrui — APPROVED** [review]

File: packages/maskbook/src/plugins/Snapshot/UI/SnapshotCard.tsx (hunk 6 above)
Comment (Jack-Works) [reply packages/maskbook/src/plugins/Snapshot/UI/SnapshotCard.tsx:10]: wait so you leave this and merged?

cc @zhouhanseng @guanbinrui

Comment (relaxgo) [reply packages/maskbook/src/plugins/Snapshot/UI/SnapshotCard.tsx:10]: should I create a new PR to fix this?

Comment (Jack-Works) [reply packages/maskbook/src/plugins/Snapshot/UI/SnapshotCard.tsx:10]: yes please

**Pull request merged**